clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -1640,6 +1640,55 @@ mlir::Value buildNeonCall(unsigned int builtinID, CIRGenFunction &cgf,
   }
 }
 
+mlir::Value CIRGenFunction::buildCommonNeonBuiltinExpr(
+    unsigned builtinID, unsigned llvmIntrinsic, unsigned altLLVMIntrinsic,
+    const char *nameHint, unsigned modifier, const CallExpr *e,
+    llvm::SmallVectorImpl<mlir::Value> &ops, cir::Address ptrOp0,
+    cir::Address ptrOp1, llvm::Triple::ArchType arch) {
+  // Get the last argument, which specifies the vector type.
+  const clang::Expr *arg = e->getArg(e->getNumArgs() - 1);
+  std::optional<llvm::APSInt> neonTypeConst =
+      arg->getIntegerConstantExpr(getContext());
+  if (!neonTypeConst)
+    return nullptr;
+
+  // Determine the type of this overloaded NEON intrinsic.
+  NeonTypeFlags neonType(neonTypeConst->getZExtValue());
+  bool isUnsigned = neonType.isUnsigned();
+  bool isQuad = neonType.isQuad();
+  const bool hasLegalHalfType = getTarget().hasLegalHalfType();
+  assert(!MissingFeatures::allowBFloatArgsAndRet());
+  // The value of allowBFloatArgsAndRet is true for AArch64, but it should
+  // come from ABI info.
+  const bool allowBFloatArgsAndRet = true;
```

Comment:
We already have `getTargetHooks().getABIInfo()` in CIRGen, please add `allowBFloatArgsAndRet` there, and instead of creating a specific missing feature for it, create one called `useTargetLoweringABIInfo` and assert on that in the `allowBFloatArgsAndRet` implementation in CIRGen.